components/editor/modules/figure/index.js

```diff
@@ -171,9 +172,51 @@ export default ({rule, subModules, TYPE}) => {
           }
         },
         schema: {
-          rules: [
-            figure
-          ]
+          blocks: {
+            [TYPE]: {
+              nodes: [
+                {
+                  types: [imageModule.TYPE],
+                  min: 1,
+                  max: 1
+                },
+                {
+                  types: [captionModule.TYPE],
+                  min: 1,
+                  max: 1
+                }
+              ]
+            },
+            normalize (change, reason, {node, index, child}) {
+              if (reason === 'child_required') {
+                change.insertNodeByKey(
+                  node.key,
+                  index,
+                  {
+                    kind: 'block',
+                    type: index === 0
+                      ? imageModule.TYPE
+                      : captionModule.TYPE
+                  }
+                )
+              }
+              if (reason === 'child_type_invalid') {
+                change.setNodeByKey(
+                  child.key,
+                  {
+                    type: index === 0
+                      ? imageModule.TYPE
+                      : captionModule.TYPE
```

Comment:
not sure but do we need to set isVoid here depending on module? Might also apply to `insertNodeByKey`. It might also be auto corrected by [slate](https://github.com/ianstormtaylor/slate/blob/master/packages/slate/src/models/schema.js#L261) but only if we define `isVoid: false` explicitly for caption afaik.

Reply:
also added `isVoid: index === 0` here